app/test/features/comments/comment_list_test.dart

```diff
@@ -0,0 +1,168 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/features/comments/widgets/comment_list_widget.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:acter/features/comments/providers/comments_providers.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:acter/features/comments/widgets/comment_list_empty_state_widget.dart';
+import 'package:mocktail/mocktail.dart';
+import '../../common/mock_data/mock_user_id.dart';
+import '../../helpers/error_helpers.dart';
+import 'mock_data/mock_comment.dart';
+import 'mock_data/mock_comments_manager.dart';
+import 'mock_data/mock_message_content.dart';
+
+void main() {
+  group('Comment List', () {
+    testWidgets('displays empty state when there are no comments',
+        (tester) async {
+      // Arrange
+      final mockCommentsManager = MockCommentsManager();
+
+      // Build the widget tree with the mocked provider
+      await tester.pumpWidget(
+        ProviderScope(
+          overrides: [
+            commentsListProvider.overrideWith((ref, manager) async => []),
+          ],
+          child: MaterialApp(
+            localizationsDelegates: const [L10n.delegate],
+            home: Scaffold(
+              body: CommentListWidget(
+                manager: mockCommentsManager, // Provide the mock manager
+              ),
+            ),
+          ),
+        ),
+      );
+
+      // Act
+      await tester.pumpAndSettle(); // Allow the widget to settle
```

Comment:
is this necessary? I'd expect the `pumpProviderWidget` to already do enough.

Reply:
Yes it is necessary. It I remove it then test give below error.

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: exactly one matching candidate
  Actual: _TypeWidgetFinder:<Found 0 widgets with type "CommentListEmptyStateWidget": []>
   Which: means none were found but one was expected

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure> (file:///Users/kumarpalsinh/Projects/Acter/a3/app/test/features/comments/comment_list_test.dart:34:7)
<asynchronous suspension>
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:189:15)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:1032:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/kumarpalsinh/Projects/Acter/a3/app/test/features/comments/comment_list_test.dart line 34
The test description was:
  displays empty state when there are no comments
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +0 -1: Comment List displays empty state when there are no comments [E]                                                                                
  Test failed. See exception logs above.
  The test description was: displays empty state when there are no comments
  

To run this test again: /Users/kumarpalsinh/Library/Flutter/flutter-sdk/bin/cache/dart-sdk/bin/dart test /Users/kumarpalsinh/Projects/Acter/a3/app/test/features/comments/comment_list_test.dart -p vm --plain-name 'Comment List displays empty state when there are no comments'
00:04 +3 -1: Some tests failed.         
```